### ISSUE TITLE
restraint doesn't always produce index.html

### ISSUE THREAD

**Issue by wizardofoz123** [beaker-project/restraint#200]
When run in standalone mode like so:

`$ restraint --job job.xml --host 1=root@<REDACTED!>`

the restraint fails to create `index.html` file in its output directory (e.g. `job.01`) on any of the following conditions:

- When External Watchdog is hit in Beaker
- When the restraint process receives `SIGINT`, `SIGTERM` or `SIGPIPE` signals

Expected behavior: `index.html` file is created.

Any tests run prior to receiving a signal or hitting the watchdog may have valid results and shouldn't be discarded. Thus, likewise, a summary of results should be created as well.

**Comment by StykMartin** [comment]
Hi @wizardofoz123 



> When run in standalone mode like so:
> 
> `$ restraint --job job.xml --host 1=root@<REDACTED!>`
> 
> the restraint fails to create `index.html` file in its output directory (e.g. `job.01`) on any of the following conditions:
> 
>     * When External Watchdog is hit in Beaker
This is quite a clear use case. EWD killed box therefore client will timeout
> 
>     * When the restraint process receives `SIGINT`, `SIGTERM` or `SIGPIPE` signals
_restraint process_. Are we speaking about `restraint` or `restraintd`? 
I believe this is the same use case as first one, but something killed it instead of turning box off.
> 
> 
> Expected behavior: `index.html` file is created.
> 
> Any tests run prior to receiving a signal or hitting the watchdog may have valid results and shouldn't be discarded. Thus, likewise, a summary of results should be created as well.

**Comment by wizardofoz123** [comment]
Hi @StykMartin

we're talking about `restraint` client process. 

Let's assume we have arbitrary tests (test1, test2, ..., testN). If test1 and test2 pass, then EWD is hit during execution of test3, shouldn't the restraint client process gracefully quite and create index.html with results that were produced prior to hitting EWD?

**Comment by wizardofoz123** [comment]
Here's my takeaway from discussing this with @cbouchar

We've confirmed that this External Watchdog Abort is from Beaker restraint.

@cbouchar suggested setting the timeout to the 2nd restraint instance (the one that actually runs tests when in standalone mode).

However, this approach has some drawbacks, most notably:
* whenever EWD is hit (e.g. machine loses connection to Beaker lab controller), the issue is still hit and we still lose data, index.html is not created
* if a watchdog times out in a virtual machine, the machine reboots for some reason and the index.html file is still not created

In other words, it's possible to workaround this problem to a degree, but it does not handle all the cases.

**Comment by cbouchar** [comment]
I've tried all the scenarios about and I always get an index.html file.  You and I need to sync-up to reproduce this.  I'm spending too much time on trying to repo.

**Comment by cbouchar** [comment]
I spent at least 2 days trying to reproduce. Spoke with the Wizard and he too was unable to reproduce.
Below is his response and he's moved onto another project.   The case he mentioned was EWD on
beaker/restraint set-up.  I don't know what Idea I had suggested that he's referring to.  I am closing
this issue.

_The case we've tried in the meeting didn't reproduce.

The idea you suggested might work good afterall.
I'll run more testing on my side, to ensure there isn't something
wrong with my env / pipeline.

The ball in on my side, I'd suggest to move onto other restraint
issues, or whatever is more pressing to you until then.

Best,
Jakub_
